mantidproject/mantid
# Replace PyEpics with CaChannel in GetLiveInstrumentValue

**Description of work.**
The PyEpics library bundled with Mantid crashes on Windows 10. This causes the algorithm `GetLiveInstrumentValue` to fail. This PR replaces the use of PyEpics in that algorithm with CaChannel. Some new instructions have been added to the wiki to describe installation of CaChannel.

**Report to:** Max at ISIS

**To test:**


You will need to be on site or on the VPN at ISIS to run these tests and ideally cycle should be running.
- First, without CaChannel installed, run this script:
```
print GetLiveInstrumentValue(Instrument='INTER',PropertyType='Run',PropertyName='TITLE')
print GetLiveInstrumentValue(Instrument='INTER',PropertyType='Block',PropertyName='Theta')
```
- You should get a sensible error message about requiring CaChannel and a link to installation instructions.
- Follow the instructions on the suggested page, restart Mantid, and try again. 
- On Mac/Ubuntu you may get a warning about caRepeater not being found. This is not ideal but only happens once per session and does not cause a failure so can be ignored. Our main users of this algorithm are reflectometry at ISIS and they are all on Windows so should not see this error.
- The current run title and Theta for INTER should be printed. You can check them against the current live values [here](http://dataweb.isis.rl.ac.uk/SeciWeb/DefaultImplement.aspx?Instrument=inter)

*There is no associated issue.*



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
